tests/integration/lldp_test.py

```diff
@@ -0,0 +1,188 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from contextlib import contextmanager
+import time
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import LLDP
+
+from .testlib import assertlib
+from .testlib import cmdlib
+from .testlib import statelib
+
+
+ETH1 = "eth1"
+
+LLDP_SYSTEM_DESC = (
+    "Summit300-48 - Version 7.4e.1 (Build 5) by Release_Master "
+    "05/27/05 04:53:11"
+)
+
+LLDP_CAPS = ["mac-bridge", "router"]
+
+
+def test_enable_lldp(eth1_up):
+    with lldp_enabled(eth1_up) as dstate:
+        assertlib.assert_state_match(dstate)
+
+
+def test_lldp_system(eth1_up):
+    with lldp_enabled(eth1_up):
+        _send_lldp_packet()
+        dstate = statelib.show_only((ETH1,))
+        lldp_config = dstate[Interface.KEY][0][LLDP.CONFIG_SUBTREE]
+        test_neighbor = _get_lldp_test(lldp_config[LLDP.NEIGHBORS_SUBTREE])
+
+        for tlv in test_neighbor:
+            if tlv["name"] == "system-name":
+                assert tlv["value"] == "Summit300-48"
+            elif tlv["name"] == "system-description":
+                assert tlv["value"] == LLDP_SYSTEM_DESC
+            elif tlv["name"] == "system-capabilities":
+                assert tlv["value"] == LLDP_CAPS
+
+
+def test_lldp_chassis(eth1_up):
+    with lldp_enabled(eth1_up):
+        _send_lldp_packet()
+        dstate = statelib.show_only((ETH1,))
+        lldp_config = dstate[Interface.KEY][0][LLDP.CONFIG_SUBTREE]
+        test_neighbor = _get_lldp_test(lldp_config[LLDP.NEIGHBORS_SUBTREE])
+
+        for tlv in test_neighbor:
+            if tlv["name"] == "chassis-id":
+                assert tlv["value"]["chassis-id"] == "00:01:30:F9:AD:A0"
```

Comment:
```suggestion
            if tlv["type"] == 1:
                assert tlv["value"] == "00:01:30:F9:AD:A0"
```

Does this make sense?